kube-client/src/config/mod.rs

```diff
@@ -271,6 +273,45 @@ impl Config {
         })
     }
 
+    /// Override configuration based on environment variables
+    ///
+    /// This is only intended for use as a debugging aid, and the specific variables and their behaviour
+    /// should **not** be considered stable across releases.
+    ///
+    /// Currently, the following overrides are supported:
+    ///
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_USER`: A Kubernetes user to impersonate, for example: `system:serviceaccount:default:foo` will impersonate the `ServiceAccount` `foo` in the `Namespace` `default`
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_GROUP`: A Kubernetes group to impersonate, multiple groups may be specified by separating them with commas
+    /// - `KUBE_RS_DEBUG_OVERRIDE_URL`: A Kubernetes cluster URL to use rather than the one specified in the config, useful for proxying traffic through `kubectl proxy`
+    #[tracing::instrument(level = "warn")]
+    pub fn apply_debug_overrides(&mut self) {
+        // Log these overrides loudly, to emphasize that this is only a debugging aid, and should not be relied upon in production
+        if let Ok(impersonate_user) = std::env::var("KUBE_RS_DEBUG_IMPERSONATE_USER") {
+            tracing::warn!(?impersonate_user, "impersonating user");
+            self.auth_info.impersonate = Some(impersonate_user);
+        }
+        if let Ok(impersonate_groups) = std::env::var("KUBE_RS_DEBUG_IMPERSONATE_GROUP") {
+            let impersonate_groups = impersonate_groups.split(',').map(str::to_string).collect();
+            tracing::warn!(?impersonate_groups, "impersonating groups");
+            self.auth_info.impersonate_groups = Some(impersonate_groups);
+        }
+        if let Ok(url) = std::env::var("KUBE_RS_DEBUG_OVERRIDE_URL") {
+            tracing::warn!(?url, "overriding cluster URL");
+            match url.parse() {
+                Ok(uri) => {
+                    self.cluster_url = uri;
+                }
+                Err(err) => {
+                    tracing::warn!(
+                        ?url,
+                        error = &err as &dyn std::error::Error,
+                        "failed to parse override cluster URL, ignoring"
+                    );
+                }
+            }
+        }
```

Comment:
you can put the secondary `proxyUrl` in the `~/.kube/config` under the `Cluster` to have it work everywhere, but i guess you want some safety here for something that is kube only?

Reply:
nevermind i see you want this as a method to be used with `kubectl proxy` rather than kube doing the proxying (which we support already)

Reply:
Yep, the latter.

There have also been cases where I have wanted to access the cluster in weird ways involving TCP reverse proxies, so this would have been helpful there as well.